core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -26,67 +33,82 @@
+    private Map<String, Integer> scalingPreferences = new HashMap<>();
+    private DockerClient dockerClient;
+
+    private static final RateLimiter AMBASSADOR_CREATION_RATE_LIMITER = RateLimiterBuilder
+            .newBuilder()
+            .withRate(6, TimeUnit.MINUTES)
+            .withConstantThroughput()
+            .build();
 
     public DockerComposeContainer(File composeFile) {
         this(composeFile, "up -d");
     }
 
-    @SuppressWarnings("WeakerAccess")
     public DockerComposeContainer(File composeFile, String command) {
-        super("dduportal/docker-compose:1.6.0");
-
-        // Create a unique identifier and tell compose
-        identifier = Base58.randomString(6).toLowerCase();
-        addEnv("COMPOSE_PROJECT_NAME", identifier);
+        this(composeFile, command, Base58.randomString(6).toLowerCase());
+    }
 
-        // Map the docker compose file into the container
-        addEnv("COMPOSE_FILE", "/compose/" + composeFile.getAbsoluteFile().getName());
-        addFileSystemBind(composeFile.getAbsoluteFile().getParentFile().getAbsolutePath(), "/compose", READ_ONLY);
+    @SuppressWarnings("WeakerAccess")
+    public DockerComposeContainer(File composeFile, String command, String identifier) {
+        this.composeFile = composeFile;
 
-        // Ensure that compose can access docker. Since the container is assumed to be running on the same machine
-        //  as the docker daemon, just mapping the docker control socket is OK.
-        // As there seems to be a problem with mapping to the /var/run directory in certain environments (e.g. CircleCI)
-        //  we map the socket file outside of /var/run, as just /docker.sock
-        addFileSystemBind("/var/run/docker.sock", "/docker.sock", READ_WRITE);
-        addEnv("DOCKER_HOST", "unix:///docker.sock");
+        // Use a unique identifier so that containers created for this compose environment can be identified
+        this.identifier = identifier;
 
-        if (command != null) {
-            setCommand(command);
-        }
+        this.dockerClient = DockerClientFactory.instance().client();
     }
 
     @Override
-    public void start() {
-
+    protected void starting(Description description) {
         final Profiler profiler = new Profiler("Docker compose container rule");
         profiler.setLogger(logger());
         profiler.start("Docker compose container startup");
 
+        applyScaling(); // scale before up, so that all scaled instances are available first for linking
+        createServices();
+        registerContainersForShutdown();
+        startAmbassadorContainers(profiler);
+
+    }
+
+
+    private void createServices() {
         // Start the docker-compose container, which starts up the services
-        super.start();
-        followOutput(new Slf4jLogConsumer(logger()), OutputFrame.OutputType.STDERR);
+        new DockerCompose(composeFile, identifier)
+                .withCommand("up -d")
+                .start();
```

Comment:
Really we're using `start()` with `docker run`-style outcomes in mind - perhaps this is a bit clumsy?